client/src/actions/authActions.js

```diff
@@ -12,18 +8,28 @@ function setCurrentUser(user) {
   };
 }
 
-function login(data) {
-  return (dispatch) => { // eslint-disable-line arrow-body-style
-    return axios.post('/api/v1/users/signin', data).then((res) => {
-      const token = res.data.token;
+/*
+const token = res.data.token;
       localStorage.setItem('jwtToken', token);
       setAuthorizationToken(token);
       dispatch(setCurrentUser(decoded));
-    });
-  };
+*/
+
+function login(data) {
+  return () => axios.post('/api/v1/users/signin', data);
```

Comment:
Why is this returning a function?

Reply:
to abstract the database request (action)...with the promise available to be worked on in the component that needs it.